snapcraft/internal/errors.py

```diff
@@ -202,14 +202,36 @@ def __init__(self, remote):
         super().__init__(remote=remote)
 
 
-class InvalidDesktopFileError(SnapcraftError):
+class MissingDesktopFileError(SnapcraftError):
 
     fmt = (
         'Failed to generate desktop file: '
-        'Invalid desktop file {filename!r}: {message}.'
-        # FIXME include how to fix each of the possible desktop file errors.
-        # https://bugs.launchpad.net/snapcraft/+bug/1727435
-        # --elopio - 2017-10-25
+        'Invalid desktop file {filename!r}: Check if desktop file matches app name or Create a new path.'
```

Comment:
I suggest making this simply `{filename!r} (defined in the app {app_name!r}) does not exist`

Reply:
I like @kyrofa's suggestion, except the parenthesis. I don't think we are using parenthesis in any of our error messages. To follow a consistent style it would be more like:
`The file {filename!r} defined in the app {app_name!r} does not exist.`.